fgpyo/fastx/__init__.py

```diff
@@ -35,11 +35,11 @@
 from typing import Type
 from typing import Union
 
-from pysam import FastxFile
-from pysam import FastxRecord
+from fgpyo._optional_dependencies import pysam
+from fgpyo._optional_dependencies import require_pysam
```

Comment:
@clintval @nh13 @tfenne If you're happy with this pattern, I can go through and update all the modules accordingly!